### PR TITLE
Overhaul the run repo in preparation for PoS launch

### PR DESCRIPTION
**Pull request by diamondhands0**
This repo will help validators get nodes up and running, and help devs make changes more effectively. Summary of everything I did:
- Upgraded all the variables for Proof of Stake and tested that they work for my own devnet node.
- Added Makefile so we do everything with a simple `make {command}`
- Cleaned up the environment variables so that all of them are now in a `base.env` file with good descriptions of what each flag does.
- Created a docker-compose.yml for each environment {local, devnet, testnet, mainnet}. This pulls in everything from base.env and explicilty overrides certain flags that are needed for the environment.
- Added a handy command to wipe the disk of each environment: `make {environment}-wipe`. The flow is you run a node for an environment using `make {environment}` and then if you want to try syncing from scratch you do `make {environment}-wipe` before you run `make {environment}` again.
- Added a `local` environment. This one is neat: If you clone a local backend+core in the parent directory and run `make local` it will run the node using your local code rather than a remote image. This could theoretically replace our reliance on n0_test and that family of scripts. I was using it to play with stuff and found it quite ergonomic, and much saner because the environment variables are cleanly organized and overridden only as needed.
- I also deleted the LICENSE anymore. I can't remember who added that or why but licenses are lame.
- Updated the README, although it could use a little more love. I'm thinking once we get testnet PoS going I can make a video where I use this repo to get a validator going on a brand new cloud machine to show them how everything works step by step.

### CODE REVIEW

File: mainnet.docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+services:
+  backend:
+    container_name: backend
+    # FIXME: We need to set this to the feature/proof-of-stake release at some point
```
Comment (lazynina) [inline mainnet.docker-compose.yml:4]:
```suggestion
```

File: testnet.docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    - API_PORT=18001
+    env_file:
+    - base.env
+    expose:
+    - "18001"
+    - "18000"
+    - "2345"
+  frontend:
+    container_name: frontend 
+    image: docker.io/desoprotocol/frontend:stable
```
Comment (lazynina) [inline testnet.docker-compose.yml:38]: stable or latest? we should just be consistent.

**Review by lazynina — CHANGES_REQUESTED** [review]

File: devnet.docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
+services:
+  backend:
+    container_name: backend
+    # FIXME: We need to set this to a more reliable release
+    # For now, just grab the latest commit from feature/proof-of-stake
+    image: docker.io/desoprotocol/backend:853e15481d6fe7a4bba97cbe79b43f9cde18b271
```
[the remark this answers is not in the document]
Comment (diamondhands0) [reply devnet.docker-compose.yml]: Using stable everywhere now

File: local.docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    ports:
+    - 18001:18001
+    - 18000:18000
+    - 2345:2345
+    environment:
+    # Values we set here OVERRIDE the base.env variables.
+    # Set the seed to a known devnet validator as an example
+    - POS_VALIDATOR_SEED=category ignore around vibrant delay cargo apart truly rabbit blue master cash
+    # For devnet, we must use a different checkpoint provider
+    - CHECKPOINT_SYNCING_PROVIDERS=https://devnet-0.deso.run
```
[the remark this answers is not in the document]
Comment (diamondhands0) [reply local.docker-compose.yml:29]: I pointed it at testnet. devnet is a little weird but I think it's fine for now.

File: mainnet.docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    # probably
+    image: docker.io/desoprotocol/backend:latest
```
[the remark this answers is not in the document]
Comment (diamondhands0) [reply mainnet.docker-compose.yml]: I put stable everywhere just to be safe. I could change it but I think it's fine.

File: mainnet.docker-compose.yml (hunk 1 above)
Comment (diamondhands0) [reply mainnet.docker-compose.yml]: Removed all fixmes.

File: testnet.docker-compose.yml (hunk 2 above)
Comment (diamondhands0) [reply testnet.docker-compose.yml:38]: Stable everywhere now.

**Review by lazynina — APPROVED** [review]
just delete the devnet file

**Comment by carry2web** [comment]
Please update the Google doc https://docs.google.com/document/d/e/2PACX-1vTBJidczUhRBhl2hTo_75o8md0RNwFguF_FeUEG_CkooN5dMLBrGyqcXvIu5efE9iAhC8ZqRr_S89Ml/pub that pointed to now merged branch feature/proof-of-stake?

**Pull request merged**